src/main/java/org/cascadebot/cascadebot/runnables/MessageReceivedRunnable.java

```diff
@@ -0,0 +1,72 @@
+package org.cascadebot.cascadebot.runnables;
+
+import com.fasterxml.jackson.core.JsonGenerationException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent;
+import org.cascadebot.cascadebot.CascadeBot;
+import org.cascadebot.cascadebot.ShutdownHandler;
+import org.cascadebot.cascadebot.data.Config;
+import org.cascadebot.cascadebot.utils.CryptUtils;
+import org.cascadebot.cascadebot.utils.PasteUtils;
+import org.cascadebot.cascadebot.utils.SerializableMessage;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.ShortBufferException;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+public class MessageReceivedRunnable implements Runnable {
+
+    private BlockingQueue<GuildMessageReceivedEvent> queue = new LinkedBlockingQueue<>();
+
+    @Override
+    public void run() {
+        while (!ShutdownHandler.SHUTDOWN_LOCK.get()) {
+            try {
+                GuildMessageReceivedEvent event = queue.take();
```

Comment:
We should probably use `poll(` which a timeout here to avoid deadlock

Reply:
I'd rather not, just in case a message isn't sent for a while, or we're testing I don't want this to stop working entirely. how would it cause deadlock as is?

Reply:
Actually it's fine because the method throws an InterruptedException.